front/pages/api/apps/[user]/[sId]/runs/index.js

```diff
@@ -131,11 +174,16 @@ export default async function handler(req, res) {
 
       const run = await runRes.json();
 
-      await app.update({
+      const updateParams = {
         savedSpecification: req.body.specification,
         savedConfig: req.body.config,
-        savedRun: run.response.run.run_id,
-      });
+      };
+
+      if (!req.body.isExecuteMode) {
```

Comment:
Had to add this param as well to prevent the runs from execute tab from overwriting the saved run -- does that sound OK to you ?